apis/loadbalancer/v1alpha1/loadbalancer_types.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2021 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+)
+
+// Known LB statuses.
+const (
+	StatusNew    = "new"
+	StatusActive = "active"
+	StatusOff    = "off"
+)
+
+// LBParameters define the desired state of a DigitalOcean LoadBalancer.
+// Most fields map directly to a LoadBalancer:
+// https://developers.digitalocean.com/documentation/v2/#load-balancers
+type LBParameters struct {
+	// Region: The unique slug identifier for the region that you wish to
+	// deploy in.
+	// +immutable
+	Region string `json:"region"`
+
+	// Algorithm: The load balancing algorithm used to determine which backend
+	// Droplet will be selected by a client.
+	// It must be either "round_robin" or "least_connections".
+	// +kubebuilder:validation:Enum=round_robin;least_connections
+	Algorithm string `json:"algorithm"`
+
+	// API Server port. It must be valid ports range (1-65535). If omitted, default value is 6443.
+	// +optional
+	// +kubebuilder:validation:Minimum=1
+	// +kubebuilder:validation:Maximum=65535
+	Port int `json:"port,omitempty"`
```

Comment:
we typically make `optional` fields pointers to distinguish between empty input and no input. Not sure if this is relevant for the DO API, but it is generally a good practice to follow for late initialization purposes. You can read more here: https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#pointer-types-and-markers

Reply:
ok, got it, will update those
thanks for your review